resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -559,15 +560,29 @@ private[spark] class MesosClusterScheduler(
       } else {
         val offer = offerOption.get
         val queuedTasks = tasks.getOrElseUpdate(offer.offerId, new ArrayBuffer[TaskInfo])
-        val task = createTaskInfo(submission, offer)
-        queuedTasks += task
-        logTrace(s"Using offer ${offer.offerId.getValue} to launch driver " +
-          submission.submissionId)
-        val newState = new MesosClusterSubmissionState(submission, task.getTaskId, offer.slaveId,
-          None, new Date(), None, getDriverFrameworkID(submission))
-        launchedDrivers(submission.submissionId) = newState
-        launchedDriversState.persist(submission.submissionId, newState)
-        afterLaunchCallback(submission.submissionId)
+        breakable {
```

Comment:
Here it needs to continue in the for loop from the catch block with next set of drivers. It cannot return from the exception since it needs to launch the other candidates, I can consider the other suggestion i.e. moving the following code into try clause. I will update the PR by moving the code into try block. Please let me know if it doesn’t make sense.